_posts/2016-12-08-underhanded-rust.en.md

```diff
@@ -0,0 +1,216 @@
+---
+layout: post
+title:  "The 2016 Underhanded Rust Contest"
+date:   2016-12-07 11:23:34 -1000
+categories: underhanded
+lang: en-US
+---
+
+The [Rust Community Team](https://community.rs) is pleased to announce the
+first annual Underhanded Rust Contest. This is a competition that tests our
+assertion that [Rust](https://www.rust-lang.org/), and our
+[ecosystem](https://crates.io/), is easy to write clean and robust software.
+Inspired by the [Underhanded C](http://www.underhanded-c.org/) and [Underhanded
+Crypto](https://underhandedcrypto.com/) contests, we want you to break our
+stuff using reasonable, easy-to-read code. We need your help to learn where our
+gaps are, and what we need to do to fix them. Can you write 100% safe Rust that
+hides a logic bug, or hide an exploit in
+[unsafe](https://doc.rust-lang.org/book/unsafe.html) Rust that passes an audit?
+Now's your chance!
```

Comment:
I think the wording of this paragraph could use some additional humility. My proposal:

> The [Rust Community Team](https://community.rs) is pleased to announce the first annual Underhanded Rust Contest, inspired by the [Underhanded C](http://www.underhanded-c.org/) and [Underhanded Crypto](https://underhandedcrypto.com/) contests. Our goal with [Rust](https://www.rust-lang.org/) is to make it easy to write trustworthy low-level software that is resistant to accidental security vulnerabilities, though less often challenged has been Rust's ability to protect against *deliberate* vulnerabilities in the face of scrutiny. This challenge is designed to put our language and [the broader Rust ecosystem](https://crates.io/) to the test, to help us learn where our blind spots are and what needs to be done to address them. In short, we want you to break our stuff using reasonable, easy-to-read code. Can you write 100% safe Rust that hides a logic bug, or hide an exploit in [unsafe](https://doc.rust-lang.org/book/unsafe.html) Rust that passes an audit? Now's your chance!